staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go

```diff
@@ -234,7 +234,6 @@ func Convert_map_to_unversioned_LabelSelector(in *map[string]string, out *LabelS
 	if in == nil {
 		return nil
 	}
-	out = new(LabelSelector)
```

Comment:
Yes this is a latent bug in the conversion code. Notice `out` is just a pointer, not a pointer to pointer, so the new LabelSelector was being dropped on the floor.